radixdlt-core/radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/tests/consensus_ledger_sync_epochs/FullNodeSyncingWithAnotherFullNodeTest.java

```diff
@@ -0,0 +1,122 @@
+import com.radixdlt.integration.distributed.simulation.monitors.consensus.ConsensusMonitors;
+import com.radixdlt.integration.distributed.simulation.monitors.ledger.LedgerMonitors;
+import com.radixdlt.integration.distributed.simulation.NetworkLatencies;
+import com.radixdlt.integration.distributed.simulation.NetworkOrdering;
+import com.radixdlt.integration.distributed.simulation.SimulationTest;
+import com.radixdlt.integration.distributed.simulation.SimulationTest.Builder;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import com.radixdlt.sync.SyncConfig;
+import org.junit.Test;
+
+/**
+ * Full (non-validator) node should be able to sync with another non-validator node that's connected to a validator.
+ */
+public class FullNodeSyncingWithAnotherFullNodeTest {
+
+	private static final ImmutableList<Integer> VALIDATORS = ImmutableList.of(0, 1);
+	private static final int NON_VALIDATOR_SYNC_NODE = 2;
+	private static final int NODE_UNDER_TEST = 3;
+	private static final int MAX_STATE_VERSION_LAG = 300;
+
+	private final Builder testBuilder;
+
+	public FullNodeSyncingWithAnotherFullNodeTest() {
+		this.testBuilder = SimulationTest.builder()
+			.numNodes(4)
+			.addressBook(ImmutableMap.of(
+				0, VALIDATORS,
+				1, VALIDATORS,
+				2, VALIDATORS,
+				// node 3 only knows of a (non-validator) node 2
+				NODE_UNDER_TEST, ImmutableList.of(NON_VALIDATOR_SYNC_NODE)
+			))
+			.networkModules(
+				NetworkOrdering.inOrder(),
+				NetworkLatencies.fixed(10)
+			)
+			.overrideWithIncorrectModule(new AbstractModule() {
+				@Provides
+				public BFTValidatorSet genesisValidatorSet(Function<Long, BFTValidatorSet> mapper) {
+					return mapper.apply(0L);
+				}
+			})
+			.pacemakerTimeout(1000)
+			.ledgerAndEpochsAndSync(
+				View.of(100),
+				(unused) -> VALIDATORS.stream().mapToInt(i -> i),
+				SyncConfig.of(200L, 10, 1000L)
+			)
+			.addTestModules(
+				ConsensusMonitors.safety(),
+				ConsensusMonitors.liveness(5, TimeUnit.SECONDS),
+				ConsensusMonitors.directParents(),
+				LedgerMonitors.consensusToLedger(),
+				LedgerMonitors.ordered(),
+				ConsensusMonitors.epochCeilingView(View.of(100))
+			);
+	}
+
+	@Test
+	public void given_a_full_node_that_is_only_conencted_to_another_full_node__it_should_sync() {
+		final var simulationTest = testBuilder.build();
+		final var runningTest = simulationTest.run();
+		final var results = runningTest.awaitCompletion();
+		assertThat(results).allSatisfy((name, err) -> assertThat(err).isEmpty());
+
+		final var testNodeCounters = runningTest.getNetwork()
+			.getSystemCounters().get(runningTest.getNetwork().getNodes().get(NODE_UNDER_TEST));
+
+		final var maxStateVersion = runningTest.getNetwork()
+			.getSystemCounters().values().stream()
+			.map(sc -> sc.get(CounterType.LEDGER_STATE_VERSION))
+			.max(Long::compareTo)
+			.get();
+
+		// make sure we're no more than MAX_STATE_VERSION_LAG versions behind the max version
```

Comment:
Would be good to have this checked during runtime as a monitor (ie full node version should never be more than X away from the highest known version)